daemon/cmd/daemon.go

```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/sirupsen/logrus"
 	"github.com/vishvananda/netlink"
 	"golang.org/x/sync/semaphore"
+	"k8s.io/apimachinery/pkg/util/wait"
```

Comment:
I think we have `pkg/backoff` in cilium for exponential backoffs, did you consider using that instead of k8s?

Reply:
@bimmlerd  Thanks for your input David! Best I can tell `pkg/backoff` expects you to manage the retry loop, thus opted for using k8s retry built-in instead which felt more appropriate for this use case.